dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/location/internal/WsLocationAdminImpl.java

```diff
@@ -335,7 +340,55 @@ protected WsLocationAdminImpl(Map<String, Object> config) {
 
         addResourcePath(bootstrapLib.getNormalizedPath());
 
-        SymbolRegistry.getRegistry().addStringSymbol(WsLocationConstants.LOC_VARIABLE_SOURCE_DIRS, (String) config.get(WsLocationConstants.LOC_VARIABLE_SOURCE_DIRS));
+        final String variableSourceDirs = (String) config.get(WsLocationConstants.LOC_VARIABLE_SOURCE_DIRS);
+        final String configDirStr = serverCfgDirStr;
+        SymbolRegistry.getRegistry().addStringSymbol(WsLocationConstants.LOC_VARIABLE_SOURCE_DIRS, variableSourceDirs);
+
+        if (bundleContext != null) {
+            //Service ranking of this hook here needs to less than service ranking of restore hook in "com.ibm.ws.config.xml.internal.SystemConfiguration.SystemConfiguration(BundleContext, SystemConfigSupport, ConfigurationAdmin). This is important in order to maintain the order of running the hooks"
+            bundleContext.registerService(CheckpointHook.class, new CheckpointHook() {
```

Comment:
This registration should be saved of for unregistration also.  Perhaps we should have WsLocationAdminImpl implement CheckpointHook and then register it as a CheckpointHook in com.ibm.ws.kernel.service.Activator.start.